cnf-tests/testsuites/e2esuite/metrics/sriovnetworkmetricsexporter.go

```diff
@@ -0,0 +1,268 @@
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	sriovtestclient "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/client"
+	sriovcluster "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	sriovnamespaces "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/namespaces"
+	sriovnetwork "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/network"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/images"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/networks"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/pods"
+
+	"github.com/prometheus/common/model"
+)
+
+const testNamespace string = "test-sriov-metrics"
+
+var sriovclient *sriovtestclient.ClientSet
+
+func init() {
+	sriovclient = sriovtestclient.New("")
+}
+
+var _ = Describe("[sriov] SR-IOV Network Metrics Exporter", func() {
+
+	var sriovCapableNodes *sriovcluster.EnabledNodes
+
+	BeforeEach(func() {
+		if discovery.Enabled() {
+			Skip("Discovery mode not supported")
+		}
+
+		restoreFeatureGates := enableMetricsExporterFeatureGate()
+		DeferCleanup(restoreFeatureGates)
+
+		By("Adding monitoring label to " + namespaces.SRIOVOperator)
+		err := sriovnamespaces.AddLabel(sriovclient, context.Background(), namespaces.SRIOVOperator, "openshift.io/cluster-monitoring", "true")
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Clean SRIOV policies and networks")
+		networks.CleanSriov(sriovclient)
+
+		By("Discover SRIOV devices")
+		sriovCapableNodes, err = sriovcluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = namespaces.Create(testNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+		namespaces.CleanPods(testNamespace, client.Client)
+	})
+
+	It("should provide the same metrics as network-metrics-daemon", func() {
+		testNode, testDevice, err := sriovCapableNodes.FindOneSriovNodeAndDevice()
+		Expect(err).ToNot(HaveOccurred())
+		By("Using device " + testDevice.Name + " on node " + testNode)
+
+		sriovNetworkNodePolicy, err := sriovnetwork.CreateSriovPolicy(
+			sriovclient, "test-metrics-", namespaces.SRIOVOperator,
+			testDevice.Name, testNode, 8,
```

Comment:
I think most of the tests use 5 vfs can we maintain the same number? It should help us if the device is a Mellanox card to have fewer reboots.

Reply:
good point, fixing